nx/include/switch/services/nfc.h

```diff
@@ -102,104 +103,123 @@ typedef enum {
     NfcMifareCommand_Store     = 0xC2,
 } NfcMifareCommand;
 
+typedef enum {
+    NfpAmiiboFlag_Valid                 = BIT(0),   ///< Initialized in system settings.
+    NfpAmiiboFlag_ApplicationAreaExists = BIT(1),   ///< Application area exists.
+} NfpAmiiboFlag;
+
+typedef enum {
+    NfpBreakType_Flush  = 0,
+    NfpBreakType_Break1 = 1,
+    NfpBreakType_Break2 = 2,
+} NfpBreakType;
+
 typedef struct {
-    u8  uuid[10];
-    u8  uuid_length;
-    u8  reserved1[0x15];
-    u32 protocol;
-    u32 tag_type;
-    u8  reserved2[0x30];
-} NX_PACKED NfpTagInfo;
+    u16 year;
+    u8 month;
+    u8 day;
+} NfpDate;
 
 typedef struct {
-    u8  uuid[10];
-    u8  uuid_length;
-    u8  reserved1[0x15];
-    u32 protocol;
-    u32 tag_type;
-    u8  reserved2[0x30];
-} NX_PACKED NfcTagInfo;
+    u8 uid[10];             ///< UUID.
+    u8 uid_length;          ///< UUID length.
+    u8 reserved[0x15];
+} NfcTagUid;
 
 typedef struct {
-    u16 last_write_year;
-    u8  last_write_month;
-    u8  last_write_day;
+    NfcTagUid uid;       ///< UUID.
+    u32 protocol;        ///< \ref NfcProtocol
+    u32 tag_type;        ///< \ref NfcTagType
+    u8 reserved[0x30];
+} NfpTagInfo;
+
+typedef struct {
+    NfcTagUid uid;       ///< UUID.
+    u32 protocol;        ///< \ref NfcProtocol
+    u32 tag_type;        ///< \ref NfcTagType
+    u8 reserved[0x30];
+} NfcTagInfo;
+
+typedef struct {
+    NfpDate last_write_date;
     u16 write_counter;
     u16 version;
     u32 application_area_size;
-    u8  reserved[0x34];
-} NX_PACKED NfpCommonInfo;
+    u8 reserved[0x34];
+} NfpCommonInfo;
 
 typedef struct {
-    u8 amiibo_id[0x8];
-    u8 reserved[0x38];
-} NX_PACKED NfpModelInfo;
+    union {
+        u8 character_id[3];
+        struct {
+            u16 game_character_id;
+            u8 character_variant;
+        } NX_PACKED;
```

Comment:
Is NX_PACKED needed here?

Reply:
Yeah, otherwise the struct would have a size of 4 bytes